$OpenBSD$

header fix
replace ftime() with gettimeofday() like emulators/fs-uae

Index: src/ints/bios.cpp
--- src/ints/bios.cpp.orig
+++ src/ints/bios.cpp
@@ -43,7 +43,6 @@
 extern bool PS1AudioCard;
 #include "parport.h"
 #include <time.h>
-#include <sys/timeb.h>
 
 #if defined(_MSC_VER)
 # pragma warning(disable:4244) /* const fmath::local::uint64_t to double possible loss of data */
@@ -3682,11 +3681,11 @@ static Bitu INT11_Handler(void) {
 
 static void BIOS_HostTimeSync() {
     /* Setup time and date */
-    struct timeb timebuffer;
-    ftime(&timebuffer);
+    struct timeval timebuffer;
+    gettimeofday(&timebuffer, NULL);
     
     struct tm *loctime;
-    loctime = localtime (&timebuffer.time);
+    loctime = localtime (&timebuffer.tv_sec);
 
     /*
     loctime->tm_hour = 23;
@@ -3705,7 +3704,7 @@ static void BIOS_HostTimeSync() {
         loctime->tm_hour*3600*1000+
         loctime->tm_min*60*1000+
         loctime->tm_sec*1000+
-        timebuffer.millitm))*(((double)PIT_TICK_RATE/65536.0)/1000.0));
+        timebuffer.tv_usec / 1000))*(((double)PIT_TICK_RATE/65536.0)/1000.0));
     mem_writed(BIOS_TIMER,ticks);
 }
 
@@ -5011,8 +5010,8 @@ bool AdapterROM_Read(Bitu address,unsigned long *size)
     return false;
 }
 
-#include "src/gui/dosbox.vga16.bmp.h"
-#include "src/gui/dosbox.cga640.bmp.h"
+#include "../src/gui/dosbox.vga16.bmp.h"
+#include "../src/gui/dosbox.cga640.bmp.h"
 
 void DrawDOSBoxLogoCGA6(unsigned int x,unsigned int y) {
     unsigned char *s = dosbox_cga640_bmp;
